src/Components/ServiceScene/LineFilterScene.tsx

```diff
@@ -1,27 +1,32 @@
 import { css } from '@emotion/css';
 import { SceneComponentProps, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
 import { Field } from '@grafana/ui';
-import { debounce, escapeRegExp } from 'lodash';
+import { debounce, escape, escapeRegExp } from 'lodash';
 import React, { ChangeEvent, KeyboardEvent } from 'react';
 import { testIds } from 'services/testIds';
 import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
 import { SearchInput } from './Breakdowns/SearchInput';
 import { LineFilterIcon } from './LineFilterIcon';
 import { getLineFilterVariable } from '../../services/variableGetters';
-import { getLineFilterCase, setLineFilterCase } from '../../services/store';
+import { getLineFilterCase, getLineFilterRegex, setLineFilterCase, setLineFilterRegex } from '../../services/store';
+import { RegexIcon, RegexInputValue } from './RegexIcon';
+import { logger } from '../../services/logger';
 
 interface LineFilterState extends SceneObjectState {
   lineFilter: string;
   caseSensitive: boolean;
+  regex: boolean;
 }
 
 export class LineFilterScene extends SceneObjectBase<LineFilterState> {
   static Component = LineFilterRenderer;
 
   constructor(state?: Partial<LineFilterState>) {
+    const caseSensitive = getLineFilterCase(false);
     super({
       lineFilter: state?.lineFilter || '',
-      caseSensitive: getLineFilterCase(false),
+      caseSensitive,
+      regex: getLineFilterRegex(false),
```

Comment:
One issue with `regex` is that we're setting it based on the stored preference and not the filter contents.